README.md

```diff
@@ -1,2 +1,95 @@
-# google-summer-of-code
-Resources for Google Summer of Code.
+# 🚀 Google Summer of Code @ WSO2
+
+Welcome to the **Google Summer of Code (GSoC) repository** for **WSO2**! This repository serves as a central hub for all things GSoC, including project ideas, past participants, and useful resources.  
+
+If you're a student looking for exciting **open-source projects to contribute to**, or a mentor interested in guiding students, you're in the right place! 🎉  
+
+---
+
+## 📌 Quick Links  
+🔹 **[📝 Ideas List](./ideas-list.md) – Start Here!**  
+🔹 [📂 Previous Projects](./projects/)  
+🔹 [🎓 Past Participants](./students/)  
+🔹 [📚 GSoC Resources](./resources/)  
+🔹 [👨‍🏫 Mentors](./mentors.md)  
+
+---
+
+## 💡 Project Ideas List  
+
+### The idea list for 2025 is now available! 🎉
+
+The **Ideas List** is the most important document in this repository. It contains well-defined project ideas that students can work on during GSoC.  
+
+✅ Each project includes:  
+- A brief overview  
+- Expected outcomes  
+- Skills required  
+- Difficulty/Commitment level  
+- Mentor(s) assigned  
+
+📢 **Check out the [Ideas List](./ideas-list.md) and start exploring!**  
+
+---
+
+## 📂 Repository Structure  
+
+This repository is organized as follows:  
+```
+google-summer-of-code/
+│── README.md                # This file
+│── ideas-list.md            # 🔥 List of project ideas
+│── mentors.md               # List of mentors & their contacts
+│── projects/                # Past GSoC projects by year
+│── students/                # Previous GSoC participants
+│── resources/               # Guides, tutorials, and links
+│── archive/                 # Older GSoC records
+│── CONTRIBUTING.md          # How to contribute
+│── CODE_OF_CONDUCT.md       # Community guidelines
+│── LICENSE                  # Repository license
+```
+
+---
+
+## 🤝 How to Contribute  
+
+We welcome contributions from students, mentors, and open-source enthusiasts! Here’s how you can help:  
+
+📌 **Students:**  
+- **Check the official GSoC website** for details on [how to apply](https://summerofcode.withgoogle.com/).  
+- Explore the **[Ideas List](./ideas-list.md)** to find a project that interests you.  
+- Research the project, understand its scope, and reach out to mentors for guidance.  
+- Start contributing to the organization’s repositories to get familiar with the codebase.  
+- Prepare a strong proposal based on the selected idea. 
+
+📌 **Mentors:**  
+- Add new project ideas to the [Ideas List](./ideas-list.md). 
+- Update mentor availability in [mentors.md](./mentors.md).  
```

Comment:
Since these instructions are for mentors, shall we say "Update your availability in"

Reply:
Fixed with [684c4b6](https://github.com/wso2/google-summer-of-code/pull/1/commits/684c4b65bfad8d9231190d8c4f3a967496ff3401)